include/pybind11/pytypes.h

```diff
@@ -613,6 +615,31 @@ struct tuple_item {
         }
     }
 };
+
+struct dict_item {
+    using key_type = object;
+
+    static object get(handle obj, handle key) {
+#if PY_MAJOR_VERSION >= 3
+        if (PyObject *result = PyDict_GetItemWithError(obj.ptr(), key.ptr())) {
```

Comment:
We could probably add a `PYBIND11_DICT_GET_ITEM_WITH_ERROR` to `common.h` to the already long list of Python 2 vs. Python 3.

Reply:
I'm fine with that so long as you think using `_PyDict_GetItemWithError` for Python 2... Yeah this will greatly simplify the code since we don't need to special-case Python 2 now.

Reply:
Discussed privately with @bstaletic, and he's also fine with this. His extra argument (besides Cython using it and PyPy also providing it): Python 2 is dead, so it won't change this (semi-private) API anymore, anyway :-)

So if you're happy with it, I think we should go for this, adding a macro in `common.h` :-)

Reply:
Oops, you already did! Great, thanks! :-)